### PR TITLE
Hack week: Lock screen scrubber setting

### LINKED ISSUE #2200: Add a setting to disable scrubbing on the lock screen
### Description

This is a feature found in several audio apps. See the thread in pdjfGa-Kp-p2 for user feedback on this.

We should be able to ignore the `MPRemoteCommandCenter.changePlaybackPositionCommand` or other callbacks.

### Screenshots or screen recording

_No response_

### Did you search for existing list?

- [X] I have searched for existing issues.

### PR DESCRIPTION
**Pull request by bjtitus**
Fixes #2200

Adds a "Lock Screen Scrubbing" setting to the general player to disable (or enable) scrubbing from the lock screen.

## To test

* Play back a podcast and verify changing the playback position in the scrubber still works
* Navigate to Profile > Settings > General and change the "Lock Screen Scrubbing" setting under "Player" from enabled to disabled
* Lock the screen again and verify changing the playback position in the scrubber no longer works

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by rviljoen** [comment]
@bjtitus Great news! Thanks for tackling this one.

Have you considered the other implementation approach mentioned here:
https://github.com/Automattic/pocket-casts-ios/issues/2200#issuecomment-2373422298

By not registering a target for `commandCenter.changePlaybackPositionCommand`, the scrubber control is completely disabled on the lock screen and dynamic island. I first noticed this implementation in the Audible app, which feels more correct to me. By just ignoring the scrub, a user who enabled the setting (maybe by accident) may be confused as to why the scrub action is ignored. Both approaches are effective though in preventing the accidental seeks.

**Comment by bjtitus** [comment]
> By not registering a target for commandCenter.changePlaybackPositionCommand, the scrubber control is completely disabled on the lock screen and dynamic island.

Oh! That's good to know. 

I just avoided that so that we didn't have to re-register the playback commands when changing this setting but I'll take another look at it. Thanks!

**Review by danielebogo — APPROVED** [review]
Nice @bjtitus ! It's working correctly! I made a test while playing and works :shipit:

File: podcasts/PlaybackManager.swift [hunk 1]
```diff
@@ -1724,6 +1704,37 @@ class PlaybackManager: ServerPlaybackDelegate {
...
+    }
+
+    @objc private func refreshRemoteCommands() {
+        let commandCenter = MPRemoteCommandCenter.shared()
+
+        commandCenter.changePlaybackPositionCommand.removeTarget(nil)
+        if !Settings.isLockScreenScrubbingDisabled { // Only perform the seek if lock screen scrubbing is enabled
+            commandCenter.changePlaybackPositionCommand.addTarget { [weak self] event -> MPRemoteCommandHandlerStatus in
+
+                guard let strongSelf = self, let _ = strongSelf.currentEpisode() else { return .noActionableNowPlayingItem }
```
Comment (danielebogo) [inline podcasts/PlaybackManager.swift]: Nit: `guard let self` should be enough here

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/67">7.79 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by rviljoen** [comment]
@bjtitus Tested the new implementation and it works flawlessly on my side. Nice one 👌🏻

**Pull request merged**